### PR TITLE
add hyper

### PR DESCRIPTION
**Pull request by PrimozGodec**
required in https://github.com/conda-forge/staged-recipes/pull/4737

### CODE REVIEW

File: recipes/hyper/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{% set name = "hyper" %}
+{% set version = "0.7.0" %}
+{% set sha256 = "069514f54231fb7b5df2fb910a114663a83306d5296f588fffcb0a9be19407fc" %}
```
Comment (dougalsutherland) [inline recipes/hyper/meta.yaml]: Builds are getting `SHA256 mismatch`; looks like the right value is `12c82eacd122a659673484c1ea0d34576430afbe5aa6b8f63fe37fcb06a2458c`.

File: recipes/hyper/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - rfc3986     >=1.1.0,<2.0
+    - brotlipy    >=0.7.0,<1.0
+
+test:
+  imports:
+    - test
+
+about:
+  home: http://hyper.rtfd.org/en/latest/
+  license: MIT
```
Comment (dougalsutherland) [inline recipes/hyper/meta.yaml]: Also add `license_family: MIT` and more importantly `license_file: LICENSE`, please.

File: recipes/hyper/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+  skip: True  # [py2k]
+
+requirements:
+  build:
+    - python
+    - setuptools
+  run:
```
Comment (dougalsutherland) [inline recipes/hyper/meta.yaml]: This one needs <code>enum34>=1.0.4,<2 &nbsp;# [py2k]</code> too.

You also might want to consider packaging the optional dependency [pycohttpparser](https://pypi.python.org/pypi/pycohttpparser/1.1.0), but can do that later if you want.

File: recipes/hyper/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
```
Comment (dougalsutherland) [inline recipes/hyper/meta.yaml]: You'll want to [add the entry point](https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#python-entry-points) too:
```
  entry_points:
    - hyper = hyper.cli:main
```

File: recipes/hyper/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  run:
+    - python
+    - h2          >=2.4,<3.0,!=2.5.0
+    - hyperframe  >=3.2,<4.0
+    - rfc3986     >=1.1.0,<2.0
+    - brotlipy    >=0.7.0,<1.0
+
+test:
+  imports:
+    - test
```
Comment (dougalsutherland) [inline recipes/hyper/meta.yaml]: Change this to `hyper`.

File: recipes/hyper/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    - python
+    - setuptools
+  run:
+    - python
+    - h2          >=2.4,<3.0,!=2.5.0
+    - hyperframe  >=3.2,<4.0
+    - rfc3986     >=1.1.0,<2.0
+    - brotlipy    >=0.7.0,<1.0
+
+test:
```
Comment (dougalsutherland) [inline recipes/hyper/meta.yaml]: Also add a test for the `hyper` command line working. This should be enough:
```
  commands:
    - hyper --help
```

**Comment by PrimozGodec** [comment]
In order to build hyper successfully, it needs an older version of h2 and hyperframe. The recipes for different version can be added to their repos but how do I make a pull request that it will be merged as a new repo?

**Comment by dougalsutherland** [comment]
You should add it as a branch to the feedstocks. I think the thing to do is to send a PR to the feedstock changing master to build that version (to make sure CIs work), then just push it as a new branch to the feedstock; I can help you through that process if you want.

**Comment by PrimozGodec** [comment]
@dougalsutherland thank you. I just didn't have an idea how to formally make PR which creates a new branch. Ok, I will make as you said.

**Comment by PrimozGodec** [comment]
@dougalsutherland do you have an idea why am I getting those conflicts?

**Comment by dougalsutherland** [comment]
@PrimozGodec  Sorry, missed that the other day. Not really sure why it failed the other day, so trying a rebuild; also needed to rebase your commits since the infrastructure for the repo changed a bit in the meantime.

**Comment by dougalsutherland** [comment]
Conda gives a better error message if you try to install the dependencies directly, instead of using the one from `conda-build`, for some reason:

```
$ conda create -n tmp python=2.7 'h2>=2.4,<3.0,!=2.5.0' 'hyperframe>=3.2,<4.0' 'rfc3986>=1.1.0,<2' 'brotlipy>=0.7.0,<1.0' 'enum34>=1.0.4,<2'
Fetching package metadata ...............
Solving package specifications: .

UnsatisfiableError: The following specifications were found to be in conflict:
  - h2 >=2.4,<3.0,!=2.5.0 -> hyperframe >=5.1.0,<6
  - hyperframe >=3.2,<4.0
Use "conda info <package>" to see the dependencies for each package.
```

So the problem is [this line](https://github.com/conda-forge/h2-feedstock/blob/b8b7dda4eafd89e3f2c7f829f5be125ec782ee90/recipe/meta.yaml#L24) in the `h2` branch; you should update that to whatever it actually should be and bump the build number, then hopefully `hyper` will build.

**Comment by dougalsutherland** [comment]
Once https://github.com/conda-forge/h2-feedstock/pull/3 is merged and built, you'll want to rebase this off current master to fix the CircleCI error.

File: recipes/hyper/meta.yaml (hunk 4 above)
Comment (dougalsutherland) [reply recipes/hyper/meta.yaml]: (Note: done.)

File: recipes/hyper/meta.yaml (hunk 6 above)
Comment (dougalsutherland) [reply recipes/hyper/meta.yaml]: (Note: done.)

File: recipes/hyper/meta.yaml (hunk 2 above)
Comment (dougalsutherland) [reply recipes/hyper/meta.yaml]: (Note: done.)

File: recipes/hyper/meta.yaml (hunk 3 above)
Comment (dougalsutherland) [reply recipes/hyper/meta.yaml]: Note: `enum34` done; pycohttpparser can be done later.

**Comment by dougalsutherland** [comment]
(Rebased for you, since CircleCI/AppVeyor at least built for `h2`.)

**Comment by dougalsutherland** [comment]
Builds locally on Mac, so...merging! Thanks @PrimozGodec. :)

**Comment by PrimozGodec** [comment]
@dougalsutherland Thank you for all your help.

**Pull request merged**